pytorch_lightning/utilities/cli.py

```diff
@@ -404,21 +404,31 @@ def __init__(
     def setup(self, trainer: Trainer, pl_module: LightningModule, stage: Optional[str] = None) -> None:
         # save the config in `setup` because (1) we want it to save regardless of the trainer function run
         # and we want to save before processes are spawned
-        log_dir = trainer.log_dir
+        log_dir = trainer.log_dir  # this broadcasts the directory
         assert log_dir is not None
+        is_global_zero = trainer.is_global_zero
         config_path = os.path.join(log_dir, self.config_filename)
-        if not self.overwrite and os.path.isfile(config_path):
-            raise RuntimeError(
-                f"{self.__class__.__name__} expected {config_path} to NOT exist. Aborting to avoid overwriting"
-                " results of a previous run. You can delete the previous config file,"
-                " set `LightningCLI(save_config_callback=None)` to disable config saving,"
-                " or set `LightningCLI(save_config_overwrite=True)` to overwrite the config file."
-            )
-        if trainer.is_global_zero:
+        fs = get_filesystem(log_dir)
+
+        if not self.overwrite:
+            # check if the file exists on rank 0
+            file_exists = fs.isfile(config_path) if is_global_zero else False
```

Comment:
@carmocca in ModelCheckpoint we have the same code except that we use `fs.exists`. Should we make it consistent? 

This special way of checking file existence with broadcast could even become a function, as we may reuse it in the future.

Reply:
Perhaps, but for a follow-up.

Do you think it should be part of checkpoint io?